### PR TITLE
Optimise and encapsulate global-nav scss

### PR DESCRIPTION
**Pull request by Caleb-Ellis**
## Done

- Reduced the amount of Vanilla dependencies to reduce the built file size (this includes removing the grid in favour of flex containers)
- Added the option to pass a config object to the `createNav()` function, so you can specify the max-width of the rows
- Renamed the global nav class names to be more descriptive
- Updated the README
- Fixed sass and JS linting errors

## QA

- Pull code and run `yarn lint` and check there are no errors
- Run `yarn watch`, wait for the file to be built and open `index.html`
- Check that the global nav functions pretty much exactly the same as the one on mongoose.ubuntu.com (I've had to make some minor changes to reduce the Vanilla dependencies)
- Remember to check in different browsers and viewpoint sizes
- Open `index.html` and pass an object with maxWidth property to the `createNav()` function:
```js
canonicalGlobalNav.createNav({
  maxWidth: '50rem'
});
```
- Check that the max-width of the global nav changes accordingly
- For extra points you can try including `dist/index.js` in different projects and running `canonicalGlobalNav.createNav()` to see how it looks

## Issues

Fixes ubuntudesign/web-squad#748

## Screenshots

Testing on the new mir-server.io homepage:

### CODE REVIEW

File: src/js/global-nav.js [hunk 1]
```diff
@@ -87,21 +89,21 @@ function createMobileDropdown(products) {
   return mobileDropdown;
 }
 
-function createProductDropdown(products) {
+function createProductDropdown(products, maxWidth) {
```
Comment (anthonydillon) [inline src/js/global-nav.js]: Does this function need `maxWidth` doesn't seem to use it?

File: src/js/global-nav.js [hunk 2]
```diff
@@ -164,60 +176,62 @@ function createProductDropdown(products) {
...
-                <a class="p-link--inverted" href=${loginItem.login}>Login&nbsp;›</a>
+            <p class="global-nav__matrix-desc">${loginItem.description}</p>
+            <ul class="global-nav__inline-list">
+              <li class="global-nav__list-item">
+                <a class="global-nav__link" href=${loginItem.login}>Login&nbsp;›</a>
               </li>
-              ${loginItem.signup ? `<li class="p-inline-list__item">
-                <a class="p-link--inverted" href=${loginItem.signup}>Sign up&nbsp;›</a>
+              ${loginItem.signup ? `<li class="global-nav__list-item">
+                <a class="global-nav__link" href=${loginItem.signup}>Sign up&nbsp;›</a>
```
Comment (anthonydillon) [inline src/js/global-nav.js]: > should be `&rsaquo`

File: src/sass/global-nav.scss [hunk 3]
```diff
@@ -0,0 +1,408 @@
...
+.global-nav {
+  @include vf-b-placeholders;
+  @include vf-b-typography;
+  @include vf-u-hide;
+  @include vf-u-margin-collapse;
+  @include vf-u-padding-collapse;
+  color: $global-nav-font-color;
+  display: flex;
+  flex-shrink: 0;
+  font-family: "Ubuntu Light", -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue", sans-serif;
```
Comment (anthonydillon) [inline src/sass/global-nav.scss]: This should come from vf-b-typography

File: src/sass/global-nav.scss [hunk 4]
```diff
@@ -0,0 +1,408 @@
...
+    text-decoration: none;
+
+    @media (max-width: $global-nav-breakpoint) {
+      padding-left: 0;
+    }
+
+    &::after {
+      @include vf-animation(transform, snap);
+      @include vf-icon-chevron($global-nav-font-color);
+      // background-image: url("data:image/svg+xml;charset=utf-8,%3Csvg xmlns='http://www.w3.org/2000/svg' height='4' width='10'%3E%3Cpath d='M3.637 3.138A26.335 26.335 0 0 1 0 0h1.541a21.242 21.242 0 0 0 1.364 1.187 16.899 16.899 0 0 0 .752.563 19.8 19.8 0 0 0 1.315.848h.054a19.8 19.8 0 0 0 1.315-.848c.398-.28.788-.583 1.169-.904.327-.275.643-.557.947-.846h1.541a26.335 26.335 0 0 1-3.637 3.138c-.519.365-.973.652-1.362.862-.39-.21-.844-.497-1.362-.862z' fill='%23fff' fill-rule='evenodd'/%3E%3C/svg%3E");
```
Comment (anthonydillon) [inline src/sass/global-nav.scss]: Please remove commented out code

File: src/sass/global-nav.scss [hunk 5]
```diff
@@ -0,0 +1,408 @@
...
+      display: block;
+      height: 1px;
+      margin-top: 2rem;
+      width: 100%;
+    }
+  }
+
+  .global-nav__strip {
+    margin: 0;
+    padding: 2rem 0 0 0;
```
Comment (anthonydillon) [inline src/sass/global-nav.scss]: `2rem 0 0`

File: src/sass/global-nav.scss [hunk 6]
```diff
@@ -0,0 +1,408 @@
...
+      display: none;
+    }
+  }
+
+  .global-nav__flex-container {
+    display: flex;
+
+    .global-nav__matrix-item {
+      border: 0;
+      padding: 0 0 2rem 0;
```
Comment (anthonydillon) [inline src/sass/global-nav.scss]: `0 0 2rem`

File: src/sass/global-nav.scss [hunk 7]
```diff
@@ -0,0 +1,408 @@
...
+    flex: 1 1 auto;
+    flex-direction: column;
+    margin: 0;
+    padding-right: 1rem;
+    width: 100%;
+  }
+
+  .global-nav__matrix-title {
+    @extend %vf-heading-4;
+    font-family: "Ubuntu Light", -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue", sans-serif;
```
Comment (anthonydillon) [inline src/sass/global-nav.scss]: Does this need font?

**Comment by Caleb-Ellis** [comment]
@anthonydillon Addressed your comments - ready for another review :+1:

**Pull request merged**